AutomationCSharpTraining/AutomationCSharpTraining/UnitTest1.cs

```diff
@@ -3,47 +3,103 @@
-            Thread.Sleep(2000);
+            UsernamePage usernamePage = homePage.EnterLoginPage();
+            PasswordPage passwordPage = usernamePage.EnterUsername1();
+            homePage = passwordPage.EnterPassword();
+            //Username is displayed in user card if logged in successfully
+            Assert.IsTrue(homePage.IsLoggedIn("ifortest"));
 
-            PasswordPage passwordPage = new PasswordPage(driver);
-            passwordPage.EnterPassword("1wsx@QAZ");
-
-            Thread.Sleep(5000);
+            driver.Quit();
+        }
+        [Test]
+        public void SuccessfullLogin2()
+        {
 
-            MailPage mailPage = new MailPage(driver);
-            
-            Assert.IsTrue(mailPage.IsLoggedIn("ifortest"));
+            HomePage homePage = new HomePage(driver);
+            homePage.URL();
+            UsernamePage usernamePage = homePage.EnterLoginPage();
+            PasswordPage passwordPage = usernamePage.EnterUsername2();
+            homePage = passwordPage.EnterPassword();
+            //Username is displayed in user card if logged in successfully
+            Assert.IsTrue(homePage.IsLoggedIn("4vtest"));      
 
-            driver.Close();
+            driver.Quit();
+        }
+        [Test]
+        public void RefreshDuringDownload()
+        {            
+            driver.Url= "https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html";
+            var DownloadButton = driver.FindElement(downloadButton);
+            //click download button
+            DownloadButton.Click();            
+            //wait till 50% appears using explicit waiter
+            var explicitWait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
+            var element = explicitWait.Until(condition =>
+            {
+                try
+                {
+                    var FiftyPercent = driver.FindElement(fiftyPercentElement);
+                    return FiftyPercent.Displayed;
+                }
+                catch (StaleElementReferenceException)
+                {
+                    return false;
+                }
+                catch (NoSuchElementException)
+                {
+                    return false;
+                }
+            }
+            );
+            //refresh page
+            driver.Navigate().Refresh();
+            //check that the page is refreshed
+            Assert.IsTrue(driver.FindElement(zeroPercentElement).Displayed);
+            driver.Quit();
+        }
+        [Test]
+        public void FilterByMinAgeMaxSalary()
+        {
+            TableSortPage tableSort = new TableSortPage(driver);
+            //choose 10 values in entries drop-down
+            tableSort.EntriesDropdown.SelectByText("10");
+            //filter
+            var FilteredList = tableSort.ByMinAgeMaxSalary(50, 200000);
+            //verify that such users exist (as an example of verification)
+            Assert.IsTrue(FilteredList.Count()>0);
```

Comment:
Add the second parameter for your Assert method. It should be message in addition to stacktrace in case test fail at this line. Assert.IsTrue(FilteredList.Count()>0, "Filtered list doesn't contain any elements")